modules/fyberBidAdapter.js

```diff
@@ -0,0 +1,415 @@
+ * @type {{defaultsParams: *, serverParamNameBySettingParamName: {referrer: string, keywords: string, appId: string, portal: string, age: string, gender: string, isSecured: (boolean|null)}, toServerParams: (function(*)), unwantedValues: *[], getUrlParams: (function(*=))}}
+ */
+const Url = {
+  defaultsParams: Object.assign({}, Helpers.defaultsQsParams, {f: CONSTANTS.DISPLAY_AD, fs: false, ref: getTopWindowReferrer()}),
+  serverParamNameBySettingParamName: {
+    referrer: 'ref',
+    keywords: 'k',
+    appId: 'aid',
+    portal: 'po',
+    age: 'a',
+    gender: 'g',
+    gdprPrivacyConsent: 'gdpr_privacy_consent',
+    consentString: 'consent_string',
+    gdprConsentData: 'gdpr_consent_data'
+  },
+  unwantedValues: ['', null, undefined],
+
+  /**
+       * Maps publisher params to server params
+       * @param params: object {k:v}
+       * @returns object {k:v}
+       */
+  toServerParams(params) {
+    const serverParams = {};
+    for (const paramName in params) {
+      if (params.hasOwnProperty(paramName) && this.serverParamNameBySettingParamName.hasOwnProperty(paramName)) {
+        serverParams[this.serverParamNameBySettingParamName[paramName]] = params[paramName];
+      } else {
+        serverParams[paramName] = params[paramName];
+      }
+    }
+
+    serverParams.isSecured = Reporter.getPageProtocol() === 'https:' || null;
+    return serverParams;
+  },
+
+  handleGDPR(params) {
+    if (params.hasOwnProperty('gdprPrivacyConsent')) {
+      if (['true', true, '1', 1].indexOf(params.gdprPrivacyConsent) !== -1) {
+        params.gdprPrivacyConsent = 1;
+      } else {
+        params.gdprPrivacyConsent = 0;
+      }
+    }
+  },
+
+  /**
+  * Prepare querty string to ad server
+  * @param params: object {k:v}
+  * @returns : object {k:v}
+  */
+  getUrlParams(params) {
+    this.handleGDPR(params);
+    const serverParams = this.toServerParams(params);
+    const toQueryString = Object.assign({}, this.defaultsParams, serverParams);
+    for (const paramName in toQueryString) {
+      if (toQueryString.hasOwnProperty(paramName) && this.unwantedValues.indexOf(toQueryString[paramName]) !== -1) {
+        delete toQueryString[paramName];
+      }
+    }
+    toQueryString.fs = params.spotType === CONSTANTS.SPOT_TYPES.INTERSTITIAL;
+
+    if (params.spotType === CONSTANTS.SPOT_TYPES.RECTANGLE) {
+      toQueryString.rw = CONSTANTS.RECTANGLE_SIZE.W;
+      toQueryString.rh = CONSTANTS.RECTANGLE_SIZE.H;
+    }
+
+    if (typeof $$PREBID_GLOBAL$$ !== 'undefined') {
+      toQueryString.bco = $$PREBID_GLOBAL$$.cbTimeout || $$PREBID_GLOBAL$$.bidderTimeout;
+    }
+
+    toQueryString.timestamp = Date.now();
+    delete toQueryString.qa;
+    return toQueryString;
+  }
+};
+
+/**
+ * Tracking pixels for events
+ * @type {{fire: (function(*=))}}
+ */
+const Tracker = {
+  /**
+  * Creates a tracking pixel
+  * @param urls: Array<String>
+  */
+  fire(urls) {
+    urls.forEach(url => url && ((new Image(1, 1)).src = encodeURI(url)));
```

Comment:
Adapters are not permitted to create their own tracking pixels.  If you want to fire a pixel, return it from getUserSyncs()